In [1]:
import pandas as pd, seaborn as sns, matplotlib.pyplot as plt

In [2]:
product_details = pd.read_csv(r"C:\Users\Boggu\Downloads\product_details.csv")
product_hierarchy = pd.read_csv(r"C:\Users\Boggu\Downloads\product_hierarchy.csv")
product_prices = pd.read_csv(r"C:\Users\Boggu\Downloads\product_prices.csv")
product_sales = pd.read_csv(r"C:\Users\Boggu\Downloads\sales (1).csv")

In [3]:
product_details

Unnamed: 0 product_id  price                      product_name  \
0            0     c4a632     13     Navy Oversized Jeans - Womens   
1            1     e83aa3     32     Black Straight Jeans - Womens   
2            2     e31d39     10      Cream Relaxed Jeans - Womens   
3            3     d5e9a6     23        Khaki Suit Jacket - Womens   
4            4     72f5d4     19       Indigo Rain Jacket - Womens   
5            5     9ec847     54      Grey Fashion Jacket - Womens   
6            6     5d267b     40            White Tee Shirt - Mens   
7            7     c8d436     10       Teal Button Up Shirt - Mens   
8            8     2a2353     57            Blue Polo Shirt - Mens   
9            9     f084eb     36           Navy Solid Socks - Mens   
10          10     b9a74d     17        White Striped Socks - Mens   
11          11     2feb6b     29  Pink Fluro Polkadot Socks - Mens   

    category_id  segment_id  style_id category_name segment_name  \
0             1           3         7        Womens        Jeans   
1             1           3         8        Womens        Jeans   
2             1           3         9        Womens        Jeans   
3             1           4        10        Womens       Jacket   
4             1           4        11        Womens       Jacket   
5             1           4        12        Womens       Jacket   
6             2           5        13          Mens        Shirt   
7             2           5        14          Mens        Shirt   
8             2           5        15          Mens        Shirt   
9             2           6        16          Mens        Socks   
10            2           6        17          Mens        Socks   
11            2           6        18          Mens        Socks   

             style_name  
0        Navy Oversized  
1        Black Straight  
2         Cream Relaxed  
3            Khaki Suit  
4           Indigo Rain  
5          Grey Fashion  
6             White Tee  
7        Teal Button Up  
8             Blue Polo  
9            Navy Solid  
10        White Striped  
11  Pink Fluro Polkadot

In [4]:
product_details.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 12 entries, 0 to 11
Data columns (total 10 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   Unnamed: 0     12 non-null     int64 
 1   product_id     12 non-null     object
 2   price          12 non-null     int64 
 3   product_name   12 non-null     object
 4   category_id    12 non-null     int64 
 5   segment_id     12 non-null     int64 
 6   style_id       12 non-null     int64 
 7   category_name  12 non-null     object
 8   segment_name   12 non-null     object
 9   style_name     12 non-null     object
dtypes: int64(5), object(5)
memory usage: 1.1+ KB


In [5]:
product_details.describe().T

count       mean        std   min    25%   50%    75%   max
Unnamed: 0    12.0   5.500000   3.605551   0.0   2.75   5.5   8.25  11.0
price         12.0  28.333333  16.069923  10.0  16.00  26.0  37.00  57.0
category_id   12.0   1.500000   0.522233   1.0   1.00   1.5   2.00   2.0
segment_id    12.0   4.500000   1.167748   3.0   3.75   4.5   5.25   6.0
style_id      12.0  12.500000   3.605551   7.0   9.75  12.5  15.25  18.0

In [6]:
product_hierarchy.set_index('Unnamed: 0', inplace = True)

In [7]:
product_hierarchy

id  parent_id           level_text level_name
Unnamed: 0                                               
0            1        NaN               Womens   Category
1            2        NaN                 Mens   Category
2            3        1.0                Jeans    Segment
3            4        1.0               Jacket    Segment
4            5        2.0                Shirt    Segment
5            6        2.0                Socks    Segment
6            7        3.0       Navy Oversized      Style
7            8        3.0       Black Straight      Style
8            9        3.0        Cream Relaxed      Style
9           10        4.0           Khaki Suit      Style
10          11        4.0          Indigo Rain      Style
11          12        4.0         Grey Fashion      Style
12          13        5.0            White Tee      Style
13          14        5.0       Teal Button Up      Style
14          15        5.0            Blue Polo      Style
15          16        6.0           Navy Solid      Style
16          17        6.0        White Striped      Style
17          18        6.0  Pink Fluro Polkadot      Style

In [8]:
product_hierarchy.info()

<class 'pandas.core.frame.DataFrame'>
Index: 18 entries, 0 to 17
Data columns (total 4 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   id          18 non-null     int64  
 1   parent_id   16 non-null     float64
 2   level_text  18 non-null     object 
 3   level_name  18 non-null     object 
dtypes: float64(1), int64(1), object(2)
memory usage: 720.0+ bytes


In [9]:
product_hierarchy.describe().T

count  mean       std  min   25%  50%    75%   max
id          18.0  9.50  5.338539  1.0  5.25  9.5  13.75  18.0
parent_id   16.0  3.75  1.693123  1.0  2.75  4.0   5.00   6.0

In [10]:
product_prices.set_index('Unnamed: 0', inplace = True)

In [11]:
product_prices

id product_id  price
Unnamed: 0                      
0            7     c4a632     13
1            8     e83aa3     32
2            9     e31d39     10
3           10     d5e9a6     23
4           11     72f5d4     19
5           12     9ec847     54
6           13     5d267b     40
7           14     c8d436     10
8           15     2a2353     57
9           16     f084eb     36
10          17     b9a74d     17
11          18     2feb6b     29

In [12]:
product_prices.info()

<class 'pandas.core.frame.DataFrame'>
Index: 12 entries, 0 to 11
Data columns (total 3 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   id          12 non-null     int64 
 1   product_id  12 non-null     object
 2   price       12 non-null     int64 
dtypes: int64(2), object(1)
memory usage: 384.0+ bytes


In [13]:
product_prices.describe().T

count       mean        std   min    25%   50%    75%   max
id      12.0  12.500000   3.605551   7.0   9.75  12.5  15.25  18.0
price   12.0  28.333333  16.069923  10.0  16.00  26.0  37.00  57.0

In [14]:
product_sales.set_index('Unnamed: 0', inplace = True)

In [15]:
product_sales

prod_id  qty  price  discount member  txn_id  \
Unnamed: 0                                                
0           c4a632    4     13        17      t  54f307   
1           5d267b    4     40        17      t  54f307   
2           b9a74d    4     17        17      t  54f307   
3           2feb6b    2     29        17      t  54f307   
4           c4a632    5     13        21      t  26cc98   
...            ...  ...    ...       ...    ...     ...   
15090       9ec847    1     54        13      t  f15ab3   
15091       2a2353    3     57        13      t  f15ab3   
15092       e83aa3    5     32         1      t  93620b   
15093       d5e9a6    2     23         1      t  93620b   
15094       5d267b    2     40         1      t  93620b   

                      start_txn_time  
Unnamed: 0                            
0            2021-02-13 01:59:43.296  
1            2021-02-13 01:59:43.296  
2            2021-02-13 01:59:43.296  
3            2021-02-13 01:59:43.296  
4           2021-01-19 01:39:00.3456  
...                              ...  
15090        2021-03-20 12:01:22.944  
15091        2021-03-20 12:01:22.944  
15092       2021-03-01 07:11:24.6624  
15093       2021-03-01 07:11:24.6624  
15094       2021-03-01 07:11:24.6624  

[15095 rows x 7 columns]

#### What was the total quantity sold for all products?

In [16]:
df = pd.merge(product_details, product_sales,left_on ='product_id' ,right_on = 'prod_id' ,how = 'left')

In [17]:
df

Unnamed: 0 product_id  price_x                      product_name  \
0               0     c4a632       13     Navy Oversized Jeans - Womens   
1               0     c4a632       13     Navy Oversized Jeans - Womens   
2               0     c4a632       13     Navy Oversized Jeans - Womens   
3               0     c4a632       13     Navy Oversized Jeans - Womens   
4               0     c4a632       13     Navy Oversized Jeans - Womens   
...           ...        ...      ...                               ...   
15090          11     2feb6b       29  Pink Fluro Polkadot Socks - Mens   
15091          11     2feb6b       29  Pink Fluro Polkadot Socks - Mens   
15092          11     2feb6b       29  Pink Fluro Polkadot Socks - Mens   
15093          11     2feb6b       29  Pink Fluro Polkadot Socks - Mens   
15094          11     2feb6b       29  Pink Fluro Polkadot Socks - Mens   

       category_id  segment_id  style_id category_name segment_name  \
0                1           3         7        Womens        Jeans   
1                1           3         7        Womens        Jeans   
2                1           3         7        Womens        Jeans   
3                1           3         7        Womens        Jeans   
4                1           3         7        Womens        Jeans   
...            ...         ...       ...           ...          ...   
15090            2           6        18          Mens        Socks   
15091            2           6        18          Mens        Socks   
15092            2           6        18          Mens        Socks   
15093            2           6        18          Mens        Socks   
15094            2           6        18          Mens        Socks   

                style_name prod_id  qty  price_y  discount member  txn_id  \
0           Navy Oversized  c4a632    4       13        17      t  54f307   
1           Navy Oversized  c4a632    5       13        21      t  26cc98   
2           Navy Oversized  c4a632    1       13        21      f  ef648d   
3           Navy Oversized  c4a632    2       13        23      t  fba96f   
4           Navy Oversized  c4a632    5       13        11      t  4e9268   
...                    ...     ...  ...      ...       ...    ...     ...   
15090  Pink Fluro Polkadot  2feb6b    5       29        21      t  e2789e   
15091  Pink Fluro Polkadot  2feb6b    4       29        12      f  8ea245   
15092  Pink Fluro Polkadot  2feb6b    2       29         5      t  f1c999   
15093  Pink Fluro Polkadot  2feb6b    2       29        12      f  15cfae   
15094  Pink Fluro Polkadot  2feb6b    2       29        14      f  72cd63   

                 start_txn_time  
0       2021-02-13 01:59:43.296  
1      2021-01-19 01:39:00.3456  
2      2021-01-27 02:18:17.1648  
3      2021-03-03 00:32:56.0544  
4      2021-01-23 14:18:54.0576  
...                         ...  
15090  2021-02-14 13:59:15.9072  
15091   2021-03-30 18:28:35.904  
15092  2021-01-10 01:21:22.6368  
15093  2021-02-09 22:01:44.6592  
15094  2021-01-13 12:47:38.6304  

[15095 rows x 17 columns]

In [18]:
df.isna().sum()

Unnamed: 0        0
product_id        0
price_x           0
product_name      0
category_id       0
segment_id        0
style_id          0
category_name     0
segment_name      0
style_name        0
prod_id           0
qty               0
price_y           0
discount          0
member            0
txn_id            0
start_txn_time    0
dtype: int64

In [19]:
df.columns

Index(['Unnamed: 0', 'product_id', 'price_x', 'product_name', 'category_id',
       'segment_id', 'style_id', 'category_name', 'segment_name', 'style_name',
       'prod_id', 'qty', 'price_y', 'discount', 'member', 'txn_id',
       'start_txn_time'],
      dtype='object')

In [20]:
df['total_quantity'] = df['price_x']*df['qty']-df['discount']

In [21]:
df

Unnamed: 0 product_id  price_x                      product_name  \
0               0     c4a632       13     Navy Oversized Jeans - Womens   
1               0     c4a632       13     Navy Oversized Jeans - Womens   
2               0     c4a632       13     Navy Oversized Jeans - Womens   
3               0     c4a632       13     Navy Oversized Jeans - Womens   
4               0     c4a632       13     Navy Oversized Jeans - Womens   
...           ...        ...      ...                               ...   
15090          11     2feb6b       29  Pink Fluro Polkadot Socks - Mens   
15091          11     2feb6b       29  Pink Fluro Polkadot Socks - Mens   
15092          11     2feb6b       29  Pink Fluro Polkadot Socks - Mens   
15093          11     2feb6b       29  Pink Fluro Polkadot Socks - Mens   
15094          11     2feb6b       29  Pink Fluro Polkadot Socks - Mens   

       category_id  segment_id  style_id category_name segment_name  \
0                1           3         7        Womens        Jeans   
1                1           3         7        Womens        Jeans   
2                1           3         7        Womens        Jeans   
3                1           3         7        Womens        Jeans   
4                1           3         7        Womens        Jeans   
...            ...         ...       ...           ...          ...   
15090            2           6        18          Mens        Socks   
15091            2           6        18          Mens        Socks   
15092            2           6        18          Mens        Socks   
15093            2           6        18          Mens        Socks   
15094            2           6        18          Mens        Socks   

                style_name prod_id  qty  price_y  discount member  txn_id  \
0           Navy Oversized  c4a632    4       13        17      t  54f307   
1           Navy Oversized  c4a632    5       13        21      t  26cc98   
2           Navy Oversized  c4a632    1       13        21      f  ef648d   
3           Navy Oversized  c4a632    2       13        23      t  fba96f   
4           Navy Oversized  c4a632    5       13        11      t  4e9268   
...                    ...     ...  ...      ...       ...    ...     ...   
15090  Pink Fluro Polkadot  2feb6b    5       29        21      t  e2789e   
15091  Pink Fluro Polkadot  2feb6b    4       29        12      f  8ea245   
15092  Pink Fluro Polkadot  2feb6b    2       29         5      t  f1c999   
15093  Pink Fluro Polkadot  2feb6b    2       29        12      f  15cfae   
15094  Pink Fluro Polkadot  2feb6b    2       29        14      f  72cd63   

                 start_txn_time  total_quantity  
0       2021-02-13 01:59:43.296              35  
1      2021-01-19 01:39:00.3456              44  
2      2021-01-27 02:18:17.1648              -8  
3      2021-03-03 00:32:56.0544               3  
4      2021-01-23 14:18:54.0576              54  
...                         ...             ...  
15090  2021-02-14 13:59:15.9072             124  
15091   2021-03-30 18:28:35.904             104  
15092  2021-01-10 01:21:22.6368              53  
15093  2021-02-09 22:01:44.6592              46  
15094  2021-01-13 12:47:38.6304              44  

[15095 rows x 18 columns]

In [22]:
a = df.groupby('product_id')['total_quantity'].sum()

In [23]:
a.sum()

1106753

#### the total quantity sold for all products is: 1106753

#### 2 What is the total generated revenue for all products before discounts?

In [24]:
df.columns

Index(['Unnamed: 0', 'product_id', 'price_x', 'product_name', 'category_id',
       'segment_id', 'style_id', 'category_name', 'segment_name', 'style_name',
       'prod_id', 'qty', 'price_y', 'discount', 'member', 'txn_id',
       'start_txn_time', 'total_quantity'],
      dtype='object')

In [25]:
df['total_revenue'] = df['price_x']*df['qty']

In [26]:
df['total_revenue'].sum()

1289453

#### the total generated revenue for all products before discounts: 1289453

#### What was the total discount amount for all products?

In [27]:
df['discount'].sum()

182700

#### the total discount amount for all products: 182700

#### How many unique transactions were there?

In [28]:
df.columns

Index(['Unnamed: 0', 'product_id', 'price_x', 'product_name', 'category_id',
       'segment_id', 'style_id', 'category_name', 'segment_name', 'style_name',
       'prod_id', 'qty', 'price_y', 'discount', 'member', 'txn_id',
       'start_txn_time', 'total_quantity', 'total_revenue'],
      dtype='object')

In [29]:
df['txn_id'].nunique()

2500

####  What is the average unique products purchased in each transaction?

In [30]:
df.columns

Index(['Unnamed: 0', 'product_id', 'price_x', 'product_name', 'category_id',
       'segment_id', 'style_id', 'category_name', 'segment_name', 'style_name',
       'prod_id', 'qty', 'price_y', 'discount', 'member', 'txn_id',
       'start_txn_time', 'total_quantity', 'total_revenue'],
      dtype='object')

In [31]:
product_sales

prod_id  qty  price  discount member  txn_id  \
Unnamed: 0                                                
0           c4a632    4     13        17      t  54f307   
1           5d267b    4     40        17      t  54f307   
2           b9a74d    4     17        17      t  54f307   
3           2feb6b    2     29        17      t  54f307   
4           c4a632    5     13        21      t  26cc98   
...            ...  ...    ...       ...    ...     ...   
15090       9ec847    1     54        13      t  f15ab3   
15091       2a2353    3     57        13      t  f15ab3   
15092       e83aa3    5     32         1      t  93620b   
15093       d5e9a6    2     23         1      t  93620b   
15094       5d267b    2     40         1      t  93620b   

                      start_txn_time  
Unnamed: 0                            
0            2021-02-13 01:59:43.296  
1            2021-02-13 01:59:43.296  
2            2021-02-13 01:59:43.296  
3            2021-02-13 01:59:43.296  
4           2021-01-19 01:39:00.3456  
...                              ...  
15090        2021-03-20 12:01:22.944  
15091        2021-03-20 12:01:22.944  
15092       2021-03-01 07:11:24.6624  
15093       2021-03-01 07:11:24.6624  
15094       2021-03-01 07:11:24.6624  

[15095 rows x 7 columns]

In [32]:
a = product_sales.groupby('txn_id')['prod_id'].nunique()

In [33]:
a.mean()

6.038

#### the average unique products purchased in each transaction: 6.038

#### What are the 25th, 50th and 75th percentile values for the revenue per transaction?

In [34]:
a = df['total_revenue'].quantile([0.25,0.50,0.75])

#### What is the average discount value per transaction?

In [35]:
df.columns

Index(['Unnamed: 0', 'product_id', 'price_x', 'product_name', 'category_id',
       'segment_id', 'style_id', 'category_name', 'segment_name', 'style_name',
       'prod_id', 'qty', 'price_y', 'discount', 'member', 'txn_id',
       'start_txn_time', 'total_quantity', 'total_revenue'],
      dtype='object')

In [36]:
df.groupby('txn_id')['discount'].sum().mean()

73.08

#### the average discount value per transaction: 73.08

#### What is the percentage split of all transactions for members vs non-members?

In [37]:
members = len(df[df['member']=='t'])
non_members = len(df[df['member']=='f'])

In [38]:
member_percentage = members/ members + non_members * 100
non_member_percentage = non_members / members + non_members * 100

In [39]:
member_percentage ,non_member_percentage 

(603401.0, 603400.6659309127)

####  What is the average revenue for member transactions and non-member transactions?

In [40]:
df[df['member'] == 't']['total_revenue'].mean()

85.75035868005739

In [41]:
df[df['member'] == 'f']['total_revenue'].mean()

84.9302287040106

### product analysis

#### What are the top 3 products by total revenue before discount?

In [42]:
df.columns

Index(['Unnamed: 0', 'product_id', 'price_x', 'product_name', 'category_id',
       'segment_id', 'style_id', 'category_name', 'segment_name', 'style_name',
       'prod_id', 'qty', 'price_y', 'discount', 'member', 'txn_id',
       'start_txn_time', 'total_quantity', 'total_revenue'],
      dtype='object')

In [43]:
df.groupby('product_name')['total_revenue'].sum().nlargest(3)

product_name
Blue Polo Shirt - Mens          217683
Grey Fashion Jacket - Womens    209304
White Tee Shirt - Mens          152000
Name: total_revenue, dtype: int64

#### What is the total quantity, revenue and discount for each segment?

In [44]:
df.groupby('segment_name').agg({'total_revenue': 'sum','discount': 'sum','qty':'sum'})

total_revenue  discount    qty
segment_name                                
Jacket               366983     45452  11385
Jeans                208350     45740  11349
Shirt                406143     46043  11265
Socks                307977     45465  11217

#### What is the top selling product for each segment?

In [45]:
df.groupby(['segment_name','product_name'])['qty'].idxmax()

segment_name  product_name                    
Jacket        Grey Fashion Jacket - Womens         6264
              Indigo Rain Jacket - Womens          5012
              Khaki Suit Jacket - Womens           3767
Jeans         Black Straight Jeans - Womens        1274
              Cream Relaxed Jeans - Womens         2521
              Navy Oversized Jeans - Womens           1
Shirt         Blue Polo Shirt - Mens              10050
              Teal Button Up Shirt - Mens          8805
              White Tee Shirt - Mens               7537
Socks         Navy Solid Socks - Mens             11321
              Pink Fluro Polkadot Socks - Mens    13839
              White Striped Socks - Mens          12608
Name: qty, dtype: int64

#### What is the total quantity, revenue and discount for each category?

In [46]:
df.groupby('category_name').agg({'total_revenue': 'sum','discount': 'sum','qty':'sum'})

total_revenue  discount    qty
category_name                                
Mens                  714120     91508  22482
Womens                575333     91192  22734

#### What is the top selling product for each category?

In [47]:
df.loc[df.groupby('category_name')['total_quantity'].idxmax()]

Unnamed: 0 product_id  price_x                  product_name  \
10229           8     2a2353       57        Blue Polo Shirt - Mens   
6329            5     9ec847       54  Grey Fashion Jacket - Womens   

       category_id  segment_id  style_id category_name segment_name  \
10229            2           5        15          Mens        Shirt   
6329             1           4        12        Womens       Jacket   

         style_name prod_id  qty  price_y  discount member  txn_id  \
10229     Blue Polo  2a2353    5       57         0      t  95a52d   
6329   Grey Fashion  9ec847    5       54         0      f  248e0e   

                 start_txn_time  total_quantity  total_revenue  
10229  2021-02-03 21:13:03.3024             285            285  
6329   2021-01-26 07:40:45.0624             270            270

####  What is the percentage split of revenue by product for each segment?

In [48]:
a = df.groupby(['segment_name', 'product_name'])['total_revenue'].sum()

In [49]:
a

segment_name  product_name                    
Jacket        Grey Fashion Jacket - Womens        209304
              Indigo Rain Jacket - Womens          71383
              Khaki Suit Jacket - Womens           86296
Jeans         Black Straight Jeans - Womens       121152
              Cream Relaxed Jeans - Womens         37070
              Navy Oversized Jeans - Womens        50128
Shirt         Blue Polo Shirt - Mens              217683
              Teal Button Up Shirt - Mens          36460
              White Tee Shirt - Mens              152000
Socks         Navy Solid Socks - Mens             136512
              Pink Fluro Polkadot Socks - Mens    109330
              White Striped Socks - Mens           62135
Name: total_revenue, dtype: int64

In [50]:
b = df.groupby('segment_name')['total_revenue'].sum()

In [51]:
b

segment_name
Jacket    366983
Jeans     208350
Shirt     406143
Socks     307977
Name: total_revenue, dtype: int64

In [52]:
percentage_split = a/b*100

In [53]:
percentage_split

segment_name  product_name                    
Jacket        Grey Fashion Jacket - Womens        57.033705
              Indigo Rain Jacket - Womens         19.451310
              Khaki Suit Jacket - Womens          23.514986
Jeans         Black Straight Jeans - Womens       58.148308
              Cream Relaxed Jeans - Womens        17.792177
              Navy Oversized Jeans - Womens       24.059515
Shirt         Blue Polo Shirt - Mens              53.597624
              Teal Button Up Shirt - Mens          8.977134
              White Tee Shirt - Mens              37.425242
Socks         Navy Solid Socks - Mens             44.325388
              Pink Fluro Polkadot Socks - Mens    35.499404
              White Striped Socks - Mens          20.175208
Name: total_revenue, dtype: float64

#### what is the percentage split of revenue by segment for each category?

In [54]:
a = df.groupby(['segment_name', 'product_name','category_id'])['total_revenue'].sum()

In [55]:
a

segment_name  product_name                      category_id
Jacket        Grey Fashion Jacket - Womens      1              209304
              Indigo Rain Jacket - Womens       1               71383
              Khaki Suit Jacket - Womens        1               86296
Jeans         Black Straight Jeans - Womens     1              121152
              Cream Relaxed Jeans - Womens      1               37070
              Navy Oversized Jeans - Womens     1               50128
Shirt         Blue Polo Shirt - Mens            2              217683
              Teal Button Up Shirt - Mens       2               36460
              White Tee Shirt - Mens            2              152000
Socks         Navy Solid Socks - Mens           2              136512
              Pink Fluro Polkadot Socks - Mens  2              109330
              White Striped Socks - Mens        2               62135
Name: total_revenue, dtype: int64

In [56]:
b = df.groupby(['category_id'])['total_revenue'].sum()

In [57]:
b

category_id
1    575333
2    714120
Name: total_revenue, dtype: int64

In [58]:
percentage_split = a/b*100

In [59]:
percentage_split 

segment_name  product_name                      category_id
Jacket        Grey Fashion Jacket - Womens      1              36.379627
              Indigo Rain Jacket - Womens       1              12.407249
              Khaki Suit Jacket - Womens        1              14.999313
Jeans         Black Straight Jeans - Womens     1              21.057718
              Cream Relaxed Jeans - Womens      1               6.443225
              Navy Oversized Jeans - Womens     1               8.712867
Shirt         Blue Polo Shirt - Mens            2              30.482692
              Teal Button Up Shirt - Mens       2               5.105584
              White Tee Shirt - Mens            2              21.284938
Socks         Navy Solid Socks - Mens           2              19.116115
              Pink Fluro Polkadot Socks - Mens  2              15.309752
              White Striped Socks - Mens        2               8.700919
Name: total_revenue, dtype: float64

#### What is the percentage split of total revenue by category?

In [60]:
a = df.groupby('category_name')['total_revenue'].sum()

In [61]:
b = a.sum()

In [62]:
percentage_split = a/b*100

In [63]:
percentage_split

category_name
Mens      55.381623
Womens    44.618377
Name: total_revenue, dtype: float64